### PR TITLE
LIXA, Laser Interferometer eXodimensional resonance Assay

### PR DESCRIPTION
**Pull request by ghost**
#### Summary
Content "LIXA: a globally unique science lab dungeon with some lore, ending in a combat gauntlet."

#### Purpose of change

LIXA is a physics lab with some significant references to actual physics, but aside from being full of little nerd tidbits, it is also a unique and interesting place to raid, alternating fights with tough military zombies, a quiet and somber exploration of an empty facility, and an unexpected genre shift into a combat gauntlet at the end.

#### Describe the solution

LIGO is a device to detect and measure gravitational waves. In our timeline, it was built in 2002, and made its first confirmed detection in 2016. The Cataclysm timeline is a little different; the first portal breach made DARPA reshuffle its priorities and change the direction of the project. The result was a device that helped lay the ground research into safe portal transit.

The four scientists who came together to run the device didn't know that, though. They were civillians. So when DARPA became XEDRA, they and the soldiers guarding the site were locked in and told to keep producing data.

Surely they haven't been down there all this time. Surely.

#### Testing

Spawn as overmap special physics_lab_LIXA to deploy.  Necessary ID cards are id_LIXA, id_LIXA_mil, and id_LIXA_2. The latter two spawn reliably once in the facility's lower levels, but up top they're still pretty scarce. There is reliable access down through the elevator shaft in the garage.

#### Additional context

NOTE: The tremendous number of lines added reflects having to make 48 different matched sets of traps, spells, EoCs, and mapgen updates, in order to accomplish a genuinely very minor parlor trick. It's just a lot of repeated stuff for now.

TO DO:

- Optimize actual encounter difficulty.
- Fix snippets involving milsec to sound more like people do.
- Spawn real road between the checkpoint and the facility
- Ensure all item groups spawn as requested, mainly clothes in lockers and tools on shelves
- Figure out congealed light terrain emitter problem.
- Create and add artifact loot for kicking the Unfolded Impossibility in the oblique angles
-  Remove debug text during unfold shift
- Add radio room, because there's mapspace in workfloor 1a and the questions of "where could a static id_LIXA_mil go" and "what happened to Lt. Kim" are the same question.

Real world source, and a genuinely fascinating thing to learn about if you have some spare time;
`www.ligo.caltech.edu`

Screenshot 2023-08-15 12 15 03 AM
Don't come in here. Seriously.

Screenshot 2023-08-15 12 16 13 AM
This is enough space for scientists to live in, right? Do they need enrichment activities?

Screenshot 2023-08-15 12 16 49 AM
That's.... probably fine.

Screenshot 2023-08-15 12 17 18 AM
_It was not fine._

Screenshot 2023-08-15 12 18 27 AM
oh hey guys what's up

### CODE REVIEW

**Comment by GuardianDll** [comment]
I'm not sure it's a good idea to compile all stuff into one it's own folder. Do you plan to put all stuff into their own folders (monsters into monster folder, map into map folder etc) afterwards?

**Comment by chichit1044** [comment]
>Do you plan to put all stuff into their own folders afterwards?

Yeah, that's usually the last step. Unfortunately, it looks like no longer works with mapgen_palettes. Which is a pain, because I can't synch the git with my files, I gotta manually put them where they go. The sacrifices one makes for nerd jokes.

**Comment by ashGlaw** [comment]
This looks awesome 👍

File: data/json/LIXA/LIXA_monsters.json [hunk 1]
```diff
@@ -0,0 +1,383 @@
...
+    "speed": 130,
+    "material": [ "flesh" ],
+    "symbol": "E",
+    "color": "light_gray",
+    "aggression": 100,
+    "morale": 100,
+    "melee_skill": 4,
+    "melee_dice": 2,
+    "melee_dice_sides": 8,
+    "melee_damage": [ { "damage_type": "pure", "amount": 0, "armor_penetration": 1 } ],
```
Comment (Maleclypse) [inline data/json/LIXA/LIXA_monsters.json:22]: I think pure damage ignores all armor already.  But I'd be interested to know if I'm wrong.

File: data/json/LIXA/LIXA_monsters.json [hunk 2]
```diff
@@ -0,0 +1,383 @@
...
+    "special_attacks": [
+      { "type": "smash", "throw_strength": 50, "cooldown": 20 },
```
Comment (Maleclypse) [inline data/json/LIXA/LIXA_monsters.json:194]: Does smash require all of the following fields?
`{ "id": "smash", "move_cost": 100, "cooldown": 8, "min_mul": 0.8, "throw_strength": 50 },`
Or at least a move cost?

**Review by Maleclypse — COMMENTED** [review]
Very cool!

File: data/json/LIXA/LIXA_monsters.json (hunk 1 above)
Comment (chichit1044) [reply data/json/LIXA/LIXA_monsters.json:22]: I will let you know when I get to that part of the testing! I actually thought the damage type here had more to do with effects on your equipment, vis. cutting your clothes or frying your electronics, but I think that's in a different spot.

File: data/json/LIXA/LIXA_monsters.json (hunk 2 above)
Comment (chichit1044) [reply data/json/LIXA/LIXA_monsters.json:194]: Genuinely no idea; I think I picked this up exactly in the middle of a recent change to the special attacks docs? I originally tried to use the hardcode, but that is one of the things that got shaken up by the Concrete Isn't Real kerfuffle.

File: data/json/Z-LIXA/LIXA/LIXA_monsters.json [hunk 3]
```diff
@@ -0,0 +1,389 @@
...
+    "vision_night": 3,
+    "harvest": "exempt",
+    "death_function": { "corpse_type": "NO_CORPSE", "message": "The %s becomes part of the endless chamber." },
+    "flags": [ "SEES", "HEARS", "WARM", "NO_BREATHE", "PUSH_MON", "HARDTOSHOOT" ],
+    "armor": { "electric": 1 }
+  },
+  {
+    "id": "LIXA_living_vector",
+    "type": "MONSTER",
+    "name": { "str": "unfolded impossibility" },
```
Comment (Procyonae) [inline data/json/Z-LIXA/LIXA/LIXA_monsters.json:214]:
```suggestion
    "name": { "str": "unfolded impossibility", "str_pl": "unfolded impossibilities" },
```

File: data/json/Z-LIXA/LIXA/LIXA_furniture_and_terrain.json [hunk 4]
```diff
@@ -0,0 +1,341 @@
...
+    "color": "light_gray",
+    "move_cost_mod": 1,
+    "required_str": -1,
+    "flags": [ "TRANSPARENT", "NOCOLLIDE" ]
+  },
+  {
+    "type": "furniture",
+    "id": "f_lixa_whiteboard",
+    "name": "whiteboard",
+    "description": "A wide space of whiteboard, for writing on with washable markers.  No one has washed it in a long time, though;  it is covered in dense scrawls of equations and waveform diagarams.",
```
Comment (Procyonae) [inline data/json/Z-LIXA/LIXA/LIXA_furniture_and_terrain.json:319]:
```suggestion
    "description": "A wide space of whiteboard, for writing on with washable markers.  No one has washed it in a long time, though; it is covered in dense scrawls of equations and waveform diagrams.",
```

**Pull request closed without merging**